src/Browscap/Data/Useragent.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * This file is part of the browscap package.
+ *
+ * Copyright (c) 1998-2017, Browser Capabilities Project
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types = 1);
+namespace Browscap\Data;
+
+/**
+ * Class Useragent
+ *
+ * @category   Browscap
+ *
+ * @author     Thomas Müller <mimmi20@live.de>
+ */
+class Useragent
+{
+    /**
+     * @var string
+     */
+    private $userAgent = '';
+
+    /**
+     * @var string[]
+     */
+    private $properties = [];
+
+    /**
+     * @var array
+     */
+    private $children = [];
+
+    /**
+     * @var string|null
+     */
+    private $platform;
+
+    /**
+     * @var string|null
+     */
+    private $engine;
+
+    /**
+     * @var string|null
+     */
+    private $device;
+
+    /**
+     * @param string      $userAgent
+     * @param string[]    $properties
+     * @param array       $children
+     * @param string|null $platform
+     * @param string|null $engine
+     * @param string|null $device
+     */
+    public function __construct(
+        string $userAgent,
+        array $properties,
+        array $children = [],
+        ?string $platform = null,
```

Comment:
Why make these optional?

Reply:
These properties are optional in an useragent section in the source files.